### PR TITLE
Feature/park info

### LINKED ISSUE #13: About
- display info about Park

### LINKED ISSUE #34: refactor router to remember the right route on load.

### LINKED ISSUE #39: Add addresses to Contact Component

### PR DESCRIPTION
**Pull request by baileydunning**
## Is this a fix or a feature?
Feature

## What is the change?
+ Display unique park info on `About`, `Location` and `Contact`
+ Map through data arrays to display multiple list items 
+ Add functionality to randomize button (note: Router is not updating)
+ Display state in banner header

## Where should the reviewer start?
+ `ParkContainer`, `ParkInfo` + Child components

## Relevant Tickets
+ Closes #13 
+ Closes #34
+ Closes #39 
+ #36 
+ #37

@saraho1123
@jaypeasee
@baileydunning

### CODE REVIEW

File: src/About/About.tsx [hunk 1]
```diff
@@ -1,12 +1,24 @@
...
-        <section className="about">
-            <h2>About GCNP</h2>
-            <p>It's the big bad ditch!!!</p>
-        </section>
-    )
+const About: React.FC<CurrentParkContainer> = props => {
+  const { currentPark } = props as CurrentParkContainer
+  const { activities } = currentPark as any
+
+  const parkActivities =  activities.map((activity: any) => {
```
Comment (saraho1123) [inline src/About/About.tsx:8]: That was a great idea to get some reps in on using .map with typescript!

File: src/App/App.tsx [hunk 2]
```diff
@@ -21,20 +21,23 @@ const App: React.FC = () => {
...
     let index = Math.floor(Math.random() * nationalParks.length)
     setParkCode(nationalParks[index].parkCode)
   }
 
   return (
     <main className="body">
-      <MainNav choosePark={choosePark}/>
+      <MainNav 
+        choosePark={choosePark}
+        generateRandomParkCode={generateRandomParkCode}/>
```
Comment (jaypeasee) [inline src/App/App.tsx:33]: We'll want to set router on to dynamically update on this eventually but not for this branch.

File: src/Banner/Banner.tsx [hunk 3]
```diff
@@ -13,7 +13,7 @@ const Banner: React.FC<CurrentParkContainer> = props => {
                 src={images[0].url}
                 alt={images[0].altText}
             /> }
-            <h1>{currentPark.fullName}</h1>
+            <h1>{currentPark.fullName}, {currentPark.states}</h1>
```
Comment (saraho1123) [inline src/Banner/Banner.tsx:16]: good idea JP!

File: src/Contact/Contact.tsx [hunk 4]
```diff
@@ -1,9 +1,35 @@
...
-function Contact() {
-    return (
-        <h2>Who ya gonna call?</h2>
-    )
+const Contact: React.FC<CurrentParkContainer> = props => {
+  console.log(props)
+  const { currentPark } = props as CurrentParkContainer
+  const { addresses } = currentPark as any
+
+  const parsedPhoneNumber = (phoneNumber: string) => {
```
Comment (saraho1123) [inline src/Contact/Contact.tsx:10]: Nice work on figuring this one out. It was tricky!

File: src/Location/Location.tsx [hunk 5]
```diff
@@ -1,9 +1,17 @@
...
 
-function Location() {
+const Location: React.FC<CurrentParkContainer> = props => {
+  const { currentPark } = props as CurrentParkContainer 
     return (
-        <h2>This is a Map!</h2>
+      <section>
+        <p><b>Driving directions to {currentPark.name}: </b>{currentPark.directionsInfo}</p>
+        <iframe 
+            src={`https://www.google.com/maps/embed?pb=!1m14!1m12!1m3!1d196281.64169563106!2d${currentPark.longitude}!3d${currentPark.latitude}!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!5e0!3m2!1sen!2sus!4v1610139788967!5m2!1sen!2sus`}
```
Comment (saraho1123) [inline src/Location/Location.tsx:10]: Thanks for teaching us how to embed maps Bailey!

File: src/Contact/Contact.tsx [hunk 6]
```diff
@@ -4,20 +4,21 @@ import './Contact.scss'
 
 const Contact: React.FC<CurrentParkContainer> = props => {
   console.log(props)
-  const { currentPark } = props as CurrentParkContainer 
-  const { contacts, emailAddresses} = currentPark as any 
+  const { currentPark } = props as CurrentParkContainer
+  const { contacts, emailAddresses } = currentPark as any
   const { addresses } = currentPark as any
 
   const parsedPhoneNumber: any = parsePhoneNumber(`+1${currentPark.contacts.phoneNumbers[0].phoneNumber}`)
```
Comment (jaypeasee) [inline src/Contact/Contact.tsx]: This is a really awesome way to format our phone numbers.

File: src/ParkInfo/ParkInfo.tsx [hunk 7]
```diff
@@ -9,42 +9,44 @@ import { RouteComponentProps } from 'react-router-dom'
...
-            </Switch>
-        </section>
-    )
+  const { currentPark } = props as CurrentParkContainer
+  return (
+    <section>
+      <Switch>
+        <Route
+          exact
+          path={`/${currentPark.parkCode}/about`}
```
Comment (saraho1123) [inline src/ParkInfo/ParkInfo.tsx:18]: Bailey, Queen of Router!

**Review by saraho1123 — APPROVED** [review]
Problem solving and learning with both of you has been a great joy! Thanks for continuing to make this a super fun project, even though it hasn't been easy!

File: src/interfaces.tsx [hunk 8]
```diff
@@ -31,6 +31,32 @@ export interface Images {
...
 
+export interface EmailAddress {
+  emailAddress: string
+}
+
+export interface PhoneNumber {
+  phoneNumber: string
+}
+
+export interface Contacts {
```
Comment (jaypeasee) [inline src/interfaces.tsx:45]: Good detailed addition to interfaces file.

**Review by jaypeasee — APPROVED** [review]

**Pull request merged**